api/v1/qdrantentity_types.go

```diff
@@ -0,0 +1,183 @@
+package v1
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/types/known/structpb"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/json"
+)
+
+//goland:noinspection GoUnusedConst
+const (
+	KindQdrantEntity     = "QdrantEntity"
+	ResourceQdrantEntity = "qdrantentities"
+)
+
+// QdrantEntitySpec defines the desired state of QdrantEntity
+type QdrantEntitySpec struct {
+	// The unique identifier of the entity (in UUID format).
+	Id string `json:"id,omitempty"`
+	// The type of the entity.
+	EntityType string `json:"entity_type,omitempty"`
+	// Timestamp when the entity was created.
+	CreatedAt metav1.Time `json:"created_at,omitempty"`
+	// Timestamp when the entity was last updated.
+	LastUpdatedAt metav1.Time `json:"last_updated_at,omitempty"`
+	// Timestamp when the entity was deleted (or is started to be deleting).
+	// If not set the entity is not deleted
+	DeletedAt metav1.Time `json:"deleted_at,omitempty"`
+	// Generic payload for this entity
+	Payload apiextensions.JSON `json:"payload,omitempty"`
```

Comment:
how about that we use `RawExtension` instead?  https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#rawextension

Reply:
We want to store a random JSON, not a k8s object here, so imo apiextensions.JSON is more appropriate, but open to check RawExtension if that makes more sense